tests/test_ag3.py

```diff
@@ -587,12 +587,21 @@ def test_snp_effects():
     assert df.iloc[674].effect == "INTRONIC"
 
 
-def test_snp_allele_frequencies():
+@pytest.mark.parametrize(
+    "cohorts",
+    [
+        {
+            "ke": "country == 'Kenya'",
+            "bf_2012_col": "country == 'Burkina Faso' and year == 2012 and species == 'coluzzii'",
+        },
+        {
+            "bf_2050_col": "country == 'Burkina Faso' and year == 2050 and species == 'coluzzii'"
+        },
+        "admin1_month",
+    ],
+)
+def test_snp_allele_frequencies(cohorts):
```

Comment:
Perhaps this is where the wizardry thingy was suggesting using separate functions? There is no point in using parametrize here, because there are completely separate code branches for each of the different values of the `cohorts` parameter. It would be clearer to have three separate test functions, e.g., named `test_snp_allele_frequencies__no_samples()`, `test_snp_allele_frequencies__str_cohorts()` and `test_snp_allele_frequencies__dict_cohorts()` or something like that.